scripts/create_changelog.py

```diff
@@ -0,0 +1,343 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+"""
+This script is used in 'Generate Changelog' workflow, defined in the 'generate_changelog.yml' file
+to automate the process of creating changelogs after each stable release.
+
+It fetches release data, extracts relevant pull request (PR) information, and generates a detailed changelog
+and add it to src/changelogs.
+The script uses the GitHub REST API to collect information about contributors, PR authors, and reviewers.
+
+Additional Note:
+- The script requires a GitHub personal access token (PAT) stored in an environment variable named `GITHUB_PAT`.
+"""
+
+import requests
+import re
+from dotenv import load_dotenv
+import os
+import argparse
+import xml.etree.ElementTree as ET
+from unidecode import unidecode
+import html
+
+load_dotenv()
+GITHUB_PAT = os.getenv('GITHUB_PAT')
+BASE_URL = r"https://api.github.com/repos/sagemath/sage"
+HEADERS = {'Authorization': f'token {GITHUB_PAT}', }
+AUTOMATED_BOTS = ['dependabot[bot]', 'github-actions', 'renovate[bot]']
+
+# Maps the github username to contributer name
+git_to_name = {}
+
+# Stores unique names of contributors
+all_contribs = set([])
+
+# Stores unique names of contributors
+first_contribs = set([])
+
+# Stores information for all the prs across all pre-releases
+# Maps tag of pre-release to its info
+all_info = {}
+
+# Stores new names which didn't previously existed in conf/contributor.xml
+new_names = []
+
+
+def get_last_name(full_name):
+    return full_name.split()[-1]
+
+
+def merge_contributors(original_file, new_contributors):
+    with open(original_file, 'r',encoding='utf-8') as f:
```

Comment:
Would you add short docstrings to functions (except simple ones), in particular explaining the arguments? They may be obvious now, but later we may forget the intention.

Reply:
Yeah sure